### PR TITLE
DROOLS-2764 : Remove dead code in AuthoringPerspective, AdministrationPerspective and Social Home Page

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2764

Bundle:
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/783
https://github.com/kiegroup/kie-wb-common/pull/1985
https://github.com/kiegroup/drools-wb/pull/894
https://github.com/kiegroup/optaplanner-wb/pull/292
https://github.com/kiegroup/jbpm-wb/pull/1182
https://github.com/kiegroup/kie-wb-distributions/pull/780
https://github.com/kiegroup/kie-docs/pull/946

### CODE REVIEW

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java [hunk 1]
```diff
@@ -1,166 +0,0 @@
...
-import org.uberfire.workbench.model.PanelDefinition;
-import org.uberfire.workbench.model.PartDefinition;
-import org.uberfire.workbench.model.PerspectiveDefinition;
-import org.uberfire.workbench.model.impl.PerspectiveDefinitionImpl;
-import org.uberfire.workbench.model.menu.MenuFactory;
-import org.uberfire.workbench.model.menu.Menus;
-
-@ApplicationScoped
-@WorkbenchPerspective(identifier = PerspectiveIds.AUTHORING, isTransient = false)
-public class DroolsAuthoringPerspective {
```
Comment (Rikkola) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java]: Authoring is replaced by `StandardEditorPerspective` and `LibraryPerspective`. Depends what you need.

**Comment by Rikkola** [comment]
Adding the removal of `AdministrationPerspective`. Looks like the commit for that is missing.

**Comment by Rikkola** [comment]
And the `AdministrationPerspective` is gone.

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java [hunk 2]
```diff
@@ -1,137 +0,0 @@
...
-import org.uberfire.workbench.model.impl.PerspectiveDefinitionImpl;
-import org.uberfire.workbench.model.menu.MenuFactory;
-import org.uberfire.workbench.model.menu.Menus;
-
-/**
- * A Perspective for Administrators
- */
-@ApplicationScoped
-@WorkbenchPerspective(identifier = PerspectiveIds.ADMINISTRATION)
-public class AdministrationPerspective {
```
Comment (Rikkola) [inline kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/AdministrationPerspective.java]: This is not used anywhere.

**Comment by Rikkola** [comment]
Jenkins please run full downstream build

**Comment by Rikkola** [comment]
@jomarko Agreed to review and test

**Comment by jomarko** [comment]
@Rikkola We miss one piece of ensemble, drools-wb:drools-wb-webapp still have dependencies to removed modules.

**Comment by Rikkola** [comment]
@jomarko Looks like I even had a commit for it, but did not push. There now.

**Comment by jomarko** [comment]
jenkins execute full downstream build

**Comment by Rikkola** [comment]
I marked these with do not merge. Just in case. I think I need to fix the docs and I'm on PTO today so I'll do have to do it next week.

**Comment by jomarko** [comment]
Jenkins please run full downstream build

**Comment by kkufova** [comment]
jenkins execute full downstream build

**Comment by Rikkola** [comment]
Jenkins please run full downstream build

**Comment by kkufova** [comment]
@Rikkola, the full downstream build does not work for some reason. Anyway, have you looked at the `pom.xml`? I cannot build the PR locally either due to the errors in `pom.xml`.

From the latest build:

> `[ERROR] 'dependencies.dependency.version' for org.kie.workbench.screens:kie-wb-common-social-home-page-api:jar is missing. @ org.drools:drools-wb-webapp:[unknown-version], /home/jenkins/workspace/kie-wb-distributions-pullrequests/upstream-repos/drools-wb/drools-wb-webapp/pom.xml, line 559, column 17`
`[ERROR] 'dependencies.dependency.version' for org.kie.workbench.screens:kie-wb-common-social-home-page-client:jar is missing. @ org.drools:drools-wb-webapp:[unknown-version], /home/jenkins/workspace/kie-wb-distributions-pullrequests/upstream-repos/drools-wb/drools-wb-webapp/pom.xml, line 563, column 17`
`[ERROR] 'dependencies.dependency.version' for org.kie.workbench.screens:kie-wb-common-social-home-page-backend:jar is missing. @ org.drools:drools-wb-webapp:[unknown-version], /home/jenkins/workspace/kie-wb-distributions-pullrequests/upstream-repos/drools-wb/drools-wb-webapp/pom.xml, line 568, column 17`

**Comment by Rikkola** [comment]
@kkufova I don't get that locally, but just rebasing and repushing everything. Maybe there is a missing commit.

**Comment by Rikkola** [comment]
Jenkins please run full downstream build

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by Rikkola** [comment]
Looks like I might have to do changes in optaplanner-wb.

**Comment by Rikkola** [comment]
All updated.

Jenkins please run full downstream build

**Comment by Rikkola** [comment]
Fails with optaplanner-wb, but that should be updated.

Jenkins please run full downstream build

**Comment by Rikkola** [comment]
Ok the optaplanner-wb failure is 13 hours old and I just tried to restart twice. Should pass when the time comes.

**Comment by Rikkola** [comment]
Am I not saying please kindly enough.

Jenkins please run full downstream build

**Comment by Rikkola** [comment]
Jenkins please run full downstream build

File: kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java (hunk 1 above)
Comment (kkufova) [reply kie-drools-wb-parent/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/perspectives/DroolsAuthoringPerspective.java]: @Rikkola, do you mean `StandaloneEditorPerspective`?

**Comment by Rikkola** [comment]
@kkufova Yep. This is because the Docks are tied to Perspectives, but used by screens. So there needs to be a bigger change on this area.

**Comment by kkufova** [comment]
I'll wait for the build with the review.

**Comment by kkufova** [comment]
jenkins execute full downstream build

**Comment by Rikkola** [comment]
Ok so the tests I listed above are picking up something real. I need to run them locally to know what is wrong. 
I'm pretty sure they are using the perspectives and can be removed, but then the question is do we need replacements for any of them.

**Comment by jhrcek** [comment]
@Rikkola This is strange. Just tried the kie-wb eap7 war from the build locally and it works fine (didn't execute the selenium tests though, but they should be using the war built during the build). The failure comes from [here](https://github.com/kiegroup/kie-wb-distributions/blob/master/kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/HomePerspective.java#L31) which means that Login page was present but after login the HomePage didn't load. Please try running the selenium tests locally (I don't want to build the whole PR ensemle now) like so and let me know if you can reproduce the failure:

```
cd kie-wb-distributions/kie-wb-tests/kie-wb-tests-gui/
mvn clean verify -Pkie-wb,wildfly11 -Dwebdriver.firefox.bin=/PATH/TO/firefox-binary
```

**Comment by kkufova** [comment]
@Rikkola, @jhrcek, I built the PR locally and run the Selenium tests. There were no failures, so I'm not sure it is related to the PR. What do you think?

**Comment by Rikkola** [comment]
@jhrcek Thanks I was just planning to do that and had no idea what commands I should use. So well timed comment.

**Comment by kkufova** [comment]
@Rikkola, @jhrcek, actually, after using the command @jhrcek suggested, the build failed (I think it has something to do with the version of the used Firefox binary, which however works fine for our internal Selenium suite). So I used just

> `mvn clean verify -Pkie-wb,wildfly11`

**Comment by jhrcek** [comment]
@kkufova @Rikkola We're using old selenium version in the upstream (2.53 - got a task to update it, but had no time to implement that change yet) which doesn't support newer versions of firefox as mentioned in the [UI tests readme](https://github.com/kiegroup/kie-wb-distributions/tree/master/kie-wb-tests/kie-wb-tests-gui).
You need to download older firefox distribution (e.g. ff46 from [here](https://ftp.mozilla.org/pub/firefox/releases/46.0b9/linux-x86_64/en-US/)) to run those tests locally.

**Comment by jhrcek** [comment]
Another possibility to consider: the last time I saw "white screen after login" was when troubleshooting swarm tests. Then my problem was too little RAM on jenkins slaves causing the tests run so slow that the UI didn't even load. Not sure if this might be the problem in this case..

**Comment by Rikkola** [comment]
Got the white screen with my latest FF. Got the older version in the link, linked to that and getting BUILD SUCCESS with this branch.

**Comment by Rikkola** [comment]
What changed? Not sure if I feel safe to merge after one green build :)

**Comment by jhrcek** [comment]
It might have been the issue with slow jenkins slaves provisioned from OpenStack (that Marek Winkler mentioned on bsig list yesterday). Let's retest again to be sure :-) 
jenkins retest this

**Comment by jhrcek** [comment]
@Rikkola This is really weird. 2 greens in a row :D And also the build was fast (1,5h). Let's merge this before we run out of luck..

**Comment by Rikkola** [comment]
Hehe. Time to merge!

**Pull request merged**